### PR TITLE
Run cirque test with GDB for backtrace

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem

Currently, Cirque is still disabled and we are running Cirque tests manually from our debug branch with GDB enabled. Enabling GDB for Cirque on master branch will help us run the regression check manually directly from master ToT with backtrace.

#### Change overview
Use GDB to run cirque test for IM and Echo for backtrace

#### Testing
How was this tested? 

Cirque is still disabled, testing is done manually with the following steps:

`Setting up cirque environment `
git submodule update --init
./scripts/tests/cirque_tests.sh bootstrap

`Run all tests `
./scripts/tests/cirque_tests.sh run_all_tests

Currently, Cirque test always failed, confirmed backtrace stack is dumped when test failed.

### CODE REVIEW

File: src/messaging/tests/echo/entrypoint.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ ot-ctl ifconfig up
 ot-ctl thread start
 
 if [ "$1" = "responder" ]; then
-    chip-echo-responder
+    gdb -return-child-result -q -ex run -ex bt chip-echo-responder
```
Comment (tcarmelveilleux) [inline src/messaging/tests/echo/entrypoint.sh:13]: This needs `-batch` on all instantiations to quit on backtrace in CI.

Also needs `-ex "thread apply all bt"` to make sure you get all stack traces

**Review by andy31415 — APPROVED** [review]

File: src/messaging/tests/echo/entrypoint.sh (hunk 1 above)
Comment (yufengwangca) [reply src/messaging/tests/echo/entrypoint.sh]: I would like to hold "thread apply all bt" until we find we do need those extra thread info. 

I found I got too many debug log on various system threads like bellow. And I am not able to easily to identify the backtrace we needed. 

Thread 5 (Thread 0x7ffff5716700 (LWP 1036688)):
#0  __libc_recv (flags=<optimized out>, len=4096, buf=0x7ffff5714c80, fd=7) at ../sysdeps/unix/sysv/linux/recv.c:28
#1  __libc_recv (fd=7, buf=0x7ffff5714c80, len=4096, flags=0) at ../sysdeps/unix/sysv/linux/recv.c:23
#2  0x000055555556991a in chip::DeviceLayer::PlatformManagerImpl::WiFIIPChangeListener() () at ../../src/platform/Linux/PlatformManagerImpl.cpp:77
#3  0x000055555556c02a in std::__invoke_impl<void, void (*)()>(std::__invoke_other, void (*&&)()) (__f=@0x55555560cc18: 0x55555556980a <chip::DeviceLayer::PlatformManagerImpl::WiFIIPChangeListener()>) at /usr/include/c++/9/bits/invoke.h:60
#4  0x000055555556bfc2 in std::__invoke<void (*)()>(void (*&&)()) (__fn=@0x55555560cc18: 0x55555556980a <chip::DeviceLayer::PlatformManagerImpl::WiFIIPChangeListener()>) at /usr/include/c++/9/bits/invoke.h:95
#5  0x000055555556bf54 in std::thread::_Invoker<std::tuple<void (*)()> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) (this=0x55555560cc18) at /usr/include/c++/9/thread:244
#6  0x000055555556bf11 in std::thread::_Invoker<std::tuple<void (*)()> >::operator()() (this=0x55555560cc18) at /usr/include/c++/9/thread:251
#7  0x000055555556bee2 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)()> > >::_M_run() (this=0x55555560cc10) at /usr/include/c++/9/thread:195
#8  0x00007ffff77d3d84 in  () at /lib/x86_64-linux-gnu/libstdc++.so.6
#9  0x00007ffff7f9b609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#10 0x00007ffff762d293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

**Comment by andy31415** [comment]
Could you add a 'how did you test' section to the summary?

**Review by woody-apple — CHANGES_REQUESTED** [review]
Per the updated template, can you update the PR here?

```
#### Problem
What is being fixed?  Examples:
* Fix crash on startup
* Fixes #12345 12345 Frobnozzle is leaky (exactly like that, so GitHub will auto-close the issue).

#### Change overview
What's in this PR

#### Testing
How was this tested? (at least one bullet point required)
    • If unit tests were added, how do they cover this issue?
    • If unit tests existed, how were they fixed/modified to prevent this in future?
    • If integration tests were added, how do they verify this change?
    • If manually tested, what platforms controller and device platforms were manually tested, and how?
    • If no testing is required, why not?
```

**Comment by yufengwangca** [comment]
> Per the updated template, can you update the PR here?
> 
> ```
> #### Problem
> What is being fixed?  Examples:
> * Fix crash on startup
> * Fixes #12345 12345 Frobnozzle is leaky (exactly like that, so GitHub will auto-close the issue).
> 
> #### Change overview
> What's in this PR
> 
> #### Testing
> How was this tested? (at least one bullet point required)
>     • If unit tests were added, how do they cover this issue?
>     • If unit tests existed, how were they fixed/modified to prevent this in future?
>     • If integration tests were added, how do they verify this change?
>     • If manually tested, what platforms controller and device platforms were manually tested, and how?
>     • If no testing is required, why not?
> ```

Done

**Review by bzbarsky-apple — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Pull request merged**